openedx/core/djangoapps/oauth_dispatch/models.py

```diff
@@ -18,7 +26,21 @@ class RestrictedApplication(models.Model):
     so that they cannot be used to call into APIs.
     """
 
-    application = models.ForeignKey(oauth2_settings.APPLICATION_MODEL, null=False)
+    application = models.OneToOneField(
+        oauth2_settings.APPLICATION_MODEL,
+        null=False,
+        related_name='restricted_application'
+    )
+
+    # a space separated list of scopes that this application can request
```

Comment:
To discuss: Don't we want the new data model and fields to go in a new table?  It doesn't need to be part of this to-be-deprecated `RestrictedApplication` concept.

Reply:
cleaned up the old code.